---
layout: page
title: Publications
permalink: /publications/
---
Google Scholar profile: [scholar.google.co.uk/citations?user=zy5V3qMAAAAJ&hl](http://scholar.google.co.uk/citations?user=zy5V3qMAAAAJ&hl)

# 2021

(50) **Nectar-feeding bats and birds show parallel molecular adaptations in sugar metabolism enzymes**  
Joshua H.T. Potter, Rosie Drinkwater, Kalina T.J. Davies, Nicolas Nesi, Marisa C.W. Lim, Laurel R. Yohe, Hai Chi, Xiaoqing Zhang, Ilya Levantis, Burton K. Lim, Christopher C. Witt, Georgia Tsagkogeorga, **Mario dos Reis**, Yang Liu, William Furey, Matthew J. Whitley, Dunja Aksentijevic, Liliana M. Dávalos, and Stephen J. Rossiter. (2021) Current Biology.  
[DOI: 10.1016/j.cub.2021.08.018](DOI:https://doi.org/10.1016/j.cub.2021.08.018)  
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1016/j.cub.2021.08.018"></div></p>

(49) **Pedigree-based and phylogenetic methods support surprising patterns of mutation rate and spectrum in the gray mouse lemur**  
C. Ryan Campbell, George P. Tiley, Jelmer W. Poelstra, Kelsie E. Hunnicutt, Peter A. Larsen, Hui-Jie Lee, Jeffrey L. Thorne, **Mario dos Reis**, and Anne D. Yoder. (2021) Heredity, 127: 233–244.  
[DOI: 10.1038/s41437-021-00446-5](https://doi.org/10.1038/s41437-021-00446-5)  
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/s41437-021-00446-5"></div></p>

(48) **Dietary diversification and specialisation in New World bats facilitated by early molecular evolution**  
Joshua H T Potter, Kalina T J Davies, Laurel R Yohe, Miluska K R Sanchez, Edgardo M Rengifo, Monika Struebig, Kim Warren, Georgia Tsagkogeorga, Burton K Lim, **Mario dos Reis**, Liliana M Dávalos, Stephen J Rossiter. (2021) Molecular Biology and Evolution, 38: 3864–3883.   
[DOI: 10.1093/molbev/msab028](https://doi.org/10.1093/molbev/msab028)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/molbev/msab028"></div></p>  

(47) **Bayesian phylogenomic dating**  
Álvarez-Carretero S, and **dos Reis M**. (2021) In: Ho S (ed.) The Molecular Evolutionary Clock: Theory
and Practice. Springer.  
[DOI: 10.1007/978-3-030-60181-2_13](https://doi.org/10.1007/978-3-030-60181-2_13)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1007/978-3-030-60181-2_13"></div></p>  

(46) **Dire wolves were the last of an ancient New World canid lineage**  
Angela R. Perri, Kieren J. Mitchell, Alice Mouton, Sandra Álvarez-Carretero, Ardern Hulme-Beaman, James Haile, Alexandra Jamieson, Julie Meachen, Audrey T. Lin, Blaine W. Schubert, Carly Ameen, Ekaterina E. Antipina, Pere Bover, Selina Brace, Alberto Carmagnini, Christian Carøe, Jose A. Samaniego Castruita, James C. Chatters, Keith Dobney, **Mario dos Reis**, Allowen Evin, Philippe Gaubert, Shyam Gopalakrishnan, Graham Gower, Holly Heiniger, Kristofer M. Helgen, Josh Kapp, Pavel A. Kosintsev, Anna Linderholm, Andrew T. Ozga, Samantha Presslee, Alexander T. Salis, Nedda F. Saremi, Colin Shew, Katherine Skerry, Dmitry E. Taranenko, Mary Thompson, Mikhail V. Sablin, Yaroslav V. Kuzmin, Matthew J. Collins, Mikkel-Holger S. Sinding, M. Thomas P. Gilbert, Anne C. Stone, Beth Shapiro, Blaire Van Valkenburgh, Robert K. Wayne, Greger Larson, Alan Cooper and Laurent A. F. Frantz (2021) Nature, 591: 87–91.  
[DOI: 10.1038/s41586-020-03082-x](https://doi.org/10.1038/s41586-020-03082-x)  
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/s41586-020-03082-x"></div></p>  

# 2020

(45) **Molecular clocks without rocks: New solutions for old problems**  
Tiley GP, Poelstra JW, **dos Reis M**, Yang Z, and Yoder, AD. (2020) Trends in Genetics, 36: 845–856.  
[DOI: 10.1016/j.tig.2020.06.002](https://doi.org/10.1016/j.tig.2020.06.002)  
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1016/j.tig.2020.06.002"></div></p>  

(44) **Phylogenomic resolution of the cetacean tree of life using target sequence capture**  
McGowen MR, Tsagkogeorga G, Álvarez-Carretero S, **dos Reis M**, Struebig M, Deaville R, Jepson PD, Jarman S, Polanowski A, Morin PA, and Rossiter SJ. (2020) Systematic Biology, 69: 479–501.  
[DOI: 10.1093/sysbio/syz068](https://doi.org/10.1093/sysbio/syz068)  
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/sysbio/syz068"></div></p>

# 2019

(43) **Phylogenomics reveals the evolutionary timing and pattern of butterflies and moths**  
Kawahara AY, Plotkin D, Espeland M, Meusemann K, Toussaint EFA, Donath A, Gimnich F, Frandsen PB, Zwick A, **dos Reis M**, Barber JR, Peters RS, Liu S, Zhou X, Mayer C, Podsiadlowski L, Storer C, Yack JE, Misof B, and Breinholt JW. (2019) Proceedings of the National Academy of Sciences, 116: 22657–22663.  
[DOI: 10.1073/pnas.1907847116](https://doi.org/10.1073/pnas.1907847116)  
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1073/pnas.1907847116"></div></p>

(42) **Bayesian estimation of species divergence times using correlated quantitative characters**  
Álvarez-Carretero S, Goswami A, Yang Z, and **dos Reis M.** (2019) Systematic Biology, 68: 967–986.
[DOI: 10.1093/sysbio/syz015](https://doi.org/10.1093/sysbio/syz015)  
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/sysbio/syz015"></div></p>  

(41) **Rapid morphological evolution in placental mammals post-dates the origin of the crown group**  
Halliday T, **dos Reis M**, Tamuri AU, Ferguson-Grow H, Yang Z, and Goswami A. (2019) Proceedings of the Royal Society B, 286: 20182418.  
[DOI: 10.1098/rspb.2018.2418](https://doi.org/10.1098/rspb.2018.2418)  
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1098/rspb.2018.2418"></div></p>

(40) **Bayesian molecular clock dating using genome-scale datasets**  
**dos Reis M** and Yang Z. (2019) In: Anisimova M (ed.) Evolutionary Genomics. Methods in Molecular Biology, vol 1910. Humana, New York, NY.   
[DOI: 10.1007/978-1-4939-9074-0_10](https://doi.org/10.1007/978-1-4939-9074-0_10)   
[Chapter and code: github.com/mariodosreis/divtime](https://github.com/mariodosreis/divtime)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1007/978-1-4939-9074-0_10"></div></p>

# 2018

(39) **Fossil-free dating**  
**dos Reis M** (2018) Nature Ecology and Evolution, 2: 771–772.  
[DOI: 10.1038/s41559-018-0532-4](https://doi.org/10.1038/s41559-018-0532-4)  
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/s41559-018-0532-4"></div></p>

(38) **Constraining uncertainty in the timescale of angiosperm evolution and the veracity of a Cretaceous Terrestrial Revolution**  
Barba-Montaya J, **dos Reis M**, Schneider H, Donoghue PJD and Yang Z. (2018) New Phytologist, 218: 819–834.  
[DOI: 10.1111/nph.15011](http://doi.org/10.1111/nph.15011)  
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1111/nph.15011"></div></p>

(37) **Using phylogenomic data to explore the effects of relaxed clocks and calibration strategies on divergence time estimation: Primates as a test case**  
**dos Reis M**, Gunnell GF, Barba-Montoya J, Wilkins A, Yang Z and Yoder AD. (2018) Systematic Biology, 67: 594–615.  
[DOI: 10.1093/sysbio/syy001](https://doi.org/10.1093/sysbio/syy001)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/sysbio/syy001"></div></p>

(36) **An evaluation of different partitioning strategies for Bayesian estimation of species divergence times**  
Angelis K, Álvarez-Carretero A, **dos Reis M** and Yang Z. (2018) Systematic Biology, 67: 61–77.  
[DOI: 10.1093/sysbio/syx061](https://dx.doi.org/10.1093/sysbio/syx061)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/sysbio/syx061"></div></p>

# 2017

(35) **A biologist’s guide to Bayesian phylogenetic analysis**  
Nascimento FF, **dos Reis M**, and Yang Z. (2017) Nature Ecology and Evolution, 1: 1446–1454.  
[DOI: 10.1038/s41559-017-0280-x](https://dx.doi.org/10.1038/s41559-017-0280-x)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/s41559-017-0280-x"></div></p>

(34) **Comparison of different strategies for using fossil calibrations to generate the time prior in Bayesian molecular clock dating**  
Barba-Montaya J, **dos Reis M** and Yang Z. (2017) Molecular Phylogenetics and Evolution, 114: 386–400.  
[DOI: 10.1016/j.ympev.2017.07.005](https://dx.doi.org/10.1016/j.ympev.2017.07.005)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1016/j.ympev.2017.07.005"></div></p>

(33) **RelTime rates collapse to a strict clock when estimating the timeline of animal diversification**  
Lozano Fernandez J, **dos Reis M**, Donoghue PCJ and Pisani D. (2017) Genome Biology and Evolution, 9: 1320–1328.  
[DOI: 10.1093/gbe/evx079](https://dx.doi.org/10.1093/gbe/evx079)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/gbe/evx079"></div></p>

(32) **Finding direction in the search for selection**  
Thiltgen G, **dos Reis M** and Goldstein RA. (2017) Journal of Molecular Evolution, 84: 39–50.  
[DOI: 10.1007/s00239-016-9765-5](https://dx.doi.org/10.1007/s00239-016-9765-5) Data: [DOI: 10.6084/m9.figshare.4284380.v1](https://dx.doi.org/10.6084/m9.figshare.4284380.v1)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1007/s00239-016-9765-5"></div></p>

# 2016

(31) **Geogenetic patterns in mouse lemurs (genus Microcebus) reveal the ghosts of Madagascar’s forests past**  
Yoder AD, Campbell CR, Blanco MB, **dos Reis M**, Ganzhorn JU, Goodman SM, Hunnicutt KE, Larsen PA, Kappeler PM, Rasoloarison RM, Ralison JM, Swofford DL and Weisrock DW. (2016) Proceedings of the National Academy of Sciences, 113: 8049–8056. **[Comments in The Washington Post,](https://href.li/?https://www.washingtonpost.com/news/speaking-of-science/wp/2016/07/18/this-absurdly-adorable-mouse-lemur-lets-scientists-travel-back-in-time/) [and National Geographic](https://href.li/?http://voices.nationalgeographic.com/2016/07/19/ridiculously-cute-mouse-lemurs-hold-key-to-madagascars-past/)**  
[DOI: 10.1073/pnas.1601081113](https://dx.doi.org/10.1073/pnas.1601081113)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1073/pnas.1601081113"></div></p>

(30) **Notes on the birth-death prior with fossil calibrations for Bayesian estimation of species divergence times**  
**dos Reis M**. (2016) Philosophical Transactions of the Royal Society B, 371: 20150128.  
[DOI: 10.1098/rstb.2015.0128](https://dx.doi.org/10.1098/rstb.2015.0128)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1098/rstb.2015.0128"></div></p>

(29) **The Interrelationships of placental mammals and the limits of phylogenetic inference**  
Tarver JE, **dos Reis M**,  Mirarab S, Moran RJ, Parker S, O’Reilly JE, King BL, O’Connell MJ, Asher RJ, Warnow T, Peterson KJ, Donoghue PCJ and Pisani D. (2016) Genome Biology and Evolution, 8: 330–344.  
[DOI: 10.1093/gbe/evv261](https://dx.doi.org/10.1093/gbe/evv261)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/gbe/evv261"></div></p>

(28) **Bayesian molecular clock dating of species divergences in the genomics era**  
**dos Reis M**, Donoghue PCJ and Yang Z. (2016) Nature Reviews Genetics, 17: 71–80.  
[DOI: 10.1038/nrg.2015.8](https://dx.doi.org/10.1038/nrg.2015.8)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1038/nrg.2015.8"></div></p>

# 2015

(27) **Uncertainty in the timing of origin of animals and the limits of precision in molecular timescales**  
**dos Reis M**, Thawornwattana Y, Angelis K, Telford M, Donoghue PCJ and Yang Z. (2015) Current Biology, 25: 2939–2950. **[Featured article](https://href.li/?http://dx.doi.org/10.1016/j.cub.2015.10.015)**  
[DOI: 10.1016/j.cub.2015.09.066](https://dx.doi.org/10.1016/j.cub.2015.09.066) Data: [DOI: 10.6084/m9.figshare.1525089](https://dx.doi.org/10.6084/m9.figshare.1525089)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1016/j.cub.2015.09.066"></div></p>

(26) **Dating tips for divergence time estimation**  
O’Reilly J, **dos Reis M** and Donoghue PCJ. (2015) Trends in Genetics, 31: 637–650.  
[DOI: 10.1016/j.tig.2015.08.001](https://dx.doi.org/10.1016/j.tig.2015.08.001)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="DOI: 10.1016/j.tig.2015.08.001"></div></p>

(25) **The impact of ancestral population size and incomplete lineage sorting on Bayesian estimation of species divergence times**  
Angelis K and **dos Reis M**. (2015) Current Zoology, 61: 874–885.  
[DOI: 10.1093/czoolo/61.5.874](https://dx.doi.org/10.1093/czoolo/61.5.874)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="DOI: 10.1093/czoolo/61.5.874"></div></p>

(24) **How to calculate the non-synonymous to synonymous rate ratio of protein-coding coding genes under the Fisher-Wright mutation-selection framework**  
**dos Reis M**. (2015) Biology Letters, 11: 20141031.  
[DOI: 10.1098/rsbl.2014.1031](https://dx.doi.org/10.1098/rsbl.2014.1031)
Data: [DOI: 10.5061/dryad.3r3q4](https://dx.doi.org/10.5061/dryad.3r3q4)
R code and examples: [DOI: 10.6084/m9.figshare.4282835.v1](https://dx.doi.org/10.6084/m9.figshare.4282835.v1)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1098/rsbl.2014.1031"></div></p>

(23) **Characterization of the uncertainty of divergence time estimation under relaxed molecular clock models using multiple loci**  
Zhu T, **dos Reis M** and Yang Z. (2015) Systematic Biology, 64: 267–280.  
[DOI: 10.1093/sysbio/syu109](https://dx.doi.org/10.1093/sysbio/syu109)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/sysbio/syu109"></div></p>

(22) **A beginner’s guide to estimating the non-synonymous to synonymous rate ratio of all protein-coding genes in a genome**  
Jeffares DC, Tomiczek B, Sojo V and **dos Reis M**. (2015) Parasite Genomics Protocols, Methods in Molecular Biology, 1201: 65–90.  
[DOI: 10.1007/978-1-4939-1438-8_4](https://dx.doi.org/10.1007/978-1-4939-1438-8_4)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1007/978-1-4939-1438-8_4"></div></p>

# 2014

(21) **Bayesian estimation of nonsynonymous/synonymous rate ratios for pairwise sequence comparisons**  
Angelis, K. **dos Reis M** and Yang Z. (2014) Molecular Biology and Evolution, 31: 1902–1913.  
[DOI: 10.1093/molbev/msu142](https://dx.doi.org/10.1093/molbev/msu142)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/molbev/msu142"></div></p>

(20) **The impact of the rate prior on Bayesian estimation of divergence times with multiple loci**  
**dos Reis M**, Zhu T and Yang Z. (2014) Systematic Biology, 63: 555–565.  
[DOI: 10.1093/sysbio/syu020](https://dx.doi.org/10.1093/sysbio/syu020)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/sysbio/syu020"></div></p>

(19) **A penalized-likelihood method to estimate the distribution of selection coefficients from phylogenetic data**  
Tamuri AU, Goldman N and **dos Reis M**. (2014) Genetics, 197: 257–271. **[Highlighted article](https://href.li/?http://www.genetics.org/content/197/1/NP.full)**  
[DOI: 10.1534/genetics.114.162263](https://dx.doi.org/10.1534/genetics.114.162263)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1534/genetics.114.162263"></div></p>

(18) **Molecular evolutionary characterization of a V1R subfamily unique to strepsirrhine Primates**  
Yoder AD, Chan LM, **dos Reis M**, Larsen PA, Campbell CR, Rasolarison R, Barrett M, Roos C, Kappeler P, Bielawski JP and Yang Z. (2014) Genome Biology and Evolution, 6: 213–227.  
[DOI: 10.1093/gbe/evu006](https://dx.doi.org/10.1093/gbe/evu006)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/gbe/evu006"></div></p>

(17) **Neither phylogenomic nor palaeontological data support a Paleogene origin of placental mammals**  
**dos Reis M**, Donoghue PCJ and Yang Z. (2014) Biology Letters, 10: 20131003. **[Comments in Nature,](https://href.li/?http://dx.doi.org/10.1038/nature.2014.14522) [The Scientist,](https://href.li/?http://www.the-scientist.com/?articles.view/articleNo/38858/title/Clocks-Versus-Rocks/) [and Phys.org](https://href.li/?http://phys.org/news/2014-01-team-fossil-only-placental-mammalian-evolution.html) - [F1000 review](https://href.li/?http://f1000.com/prime/718241049)**  
[DOI: 10.1098/rsbl.2013.1003](https://dx.doi.org/10.1098/rsbl.2013.1003)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1098/rsbl.2013.1003"></div></p>

# 2013

(16) **Why do more divergent sequences produce smaller nonsynonymous/synonymous rate ratios in pairwise sequence comparisons?**  
**dos Reis M** and Yang Z. (2013) Genetics, 195: 195–204.  
[DOI: 10.1534/genetics.113.152025](https://dx.doi.org/10.1534/genetics.113.152025)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="DOI: 10.1534/genetics.113.152025"></div></p>

(15) **The unbearable uncertainty of Bayesian divergence time estimation**  
**dos Reis M** and Yang Z. (2013) Journal of Systematics and Evolution, 51: 30–43.  
[DOI: 10.1111/j.1759-6831.2012.00236.x](https://dx.doi.org/10.1111/j.1759-6831.2012.00236.x)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1111/j.1759-6831.2012.00236.x"></div></p>

# 2012

(14) **Phylogenomic datasets provide both precision and accuracy in estimating the timescale of placental mammal phylogeny**  
**dos Reis M**, Inoue J, Hasegawa M, Asher RJ, Donoghue PCJ and Yang Z. (2012) Proceedings of the Royal Society B., 279: 3491–3500. **[Timetrees (nexus)](https://href.li/?http://people.cryst.bbk.ac.uk/%7Efdosr01/mamdiv/trees.zip) - [274 species timetree (pdf)](https://href.li/?http://abacus.gene.ucl.ac.uk/divtimes/figs/mammal-large-timetree.pdf) - [Data](https://href.li/?http://abacus.gene.ucl.ac.uk/ziheng/data.html) - [F1000 review](https://href.li/?http://f1000.com/prime/717598023)**  
[DOI: 10.1098/rspb.2012.0683](https://dx.doi.org/10.1098/rspb.2012.0683)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1098/rspb.2012.0683"></div></p>

(13) **Estimating the distribution of selection coefficients from phylogenetic data using sitewise mutation-selection models**  
Tamuri AU, **dos Reis M** and Goldstein RA. (2012) Genetics, 190: 1101–1115.  
[DOI: 10.1534/genetics.111.136432](https://dx.doi.org/10.1534/genetics.111.136432)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1534/genetics.111.136432"></div></p>

# 2011

(12) **An exceptionally high nucleotide and haplotype diversity and a signature of positive selection for the eIF4E resistance gene in barley are revealed by allele mining and phylogenetic analyses of natural populations**  
Hofinger BJ, Russell JR, Bass CG, Baldwin T, **dos Reis M**, Hedley PE, Li Y, Macaulay M, Waugh R, Hammond-Kosack KE and Kanyuka K. (2011) Molecular Ecology, 20: 3653–3668.  
[DOI: 10.1111/j.1365-294X.2011.05201.x](https://dx.doi.org/10.1111/j.1365-294X.2011.05201.x)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1111/j.1365-294X.2011.05201.x"></div></p>

(11) **Approximate likelihood calculation on a phylogeny for Bayesian estimation of divergence times**  
**dos Reis M** and Yang Z. (2011) Molecular Biology and Evolution, 28: 2161–2172. **[R code](https://href.li/?http://people.cryst.bbk.ac.uk/%7Efdosr01/R/approx.R)**  
[DOI: 10.1093/molbev/msr045](https://dx.doi.org/10.1093/molbev/msr045)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/molbev/msr045"></div></p>

(10) **Charting the host adaptation of influenza viruses**  
**dos Reis M**, Tamuri AU, Hay AJ and Goldstein RA. (2011) Molecular Biology and Evolution, 28: 1755–1767. **[Featured in Evolutionary Analysis](https://href.li/?http://www.amazon.co.uk/Evolutionary-Analysis-Jon-C-Herron/dp/0321616677/ref=sr_1_1?ie=UTF8&qid=1408379766&sr=8-1&keywords=evolutionary+analysis)**  
[DOI: 10.1093/molbev/msq317](https://dx.doi.org/10.1093/molbev/msq317)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/molbev/msq317"></div></p>

(9) **Statistical properties of the branch-site test of positive selection**  
Yang Z and **dos Reis M**. (2011) Molecular Biology and Evolution, 28: 1217–1228.  
[DOI: 10.1093/molbev/msq303](https://dx.doi.org/10.1093/molbev/msq303)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/molbev/msq303"></div></p>

# 2010

(8) **Analysis of the Albumin/alpha-Fetoprotein/Afamin/Group specific component gene family in the context of zebrafish liver differentation**  
Noël ES, **dos Reis M**, Arain Z and Ober EA. (2010) Gene Expression Patterns, 10: 237–243.  
[DOI: 10.1016/j.gep.2010.05.002](https://dx.doi.org/10.1016/j.gep.2010.05.002)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1016/j.gep.2010.05.002"></div></p>

# 2009

(7) **Identifying changes in selective constraints: host shifts in influenza**  
Tamuri AU, **dos Reis M**, Hay AJ and Goldstein RA. (2009) PLoS Computational Biology, 5: e1000564.  
[DOI: 10.1371/journal.pcbi.1000564](https://dx.doi.org/10.1371/journal.pcbi.1000564)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1371/journal.pcbi.1000564"></div></p>

(6) **Using non-homogeneous models to identify host shift events: application to the origin of the 1918 ‘Spanish’ influenza virus**  
**dos Reis M**, Hay AJ and Goldstein RA. (2009) Journal of Molecular Evolution, 69: 333–345. **[Additional data](https://href.li/?http://people.cryst.bbk.ac.uk/%7Efdosr01/1918flu/1918flu.zip) - [Featured in Evolutionary Analysis](https://href.li/?http://www.amazon.co.uk/Evolutionary-Analysis-Jon-C-Herron/dp/0321616677/ref=sr_1_1?ie=UTF8&qid=1408379766&sr=8-1&keywords=evolutionary+analysis)**  
[DOI: 10.1007/s00239-009-9282-x](https://dx.doi.org/10.1007/s00239-009-9282-x)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1007/s00239-009-9282-x"></div></p>

(5) **Estimating translational selection in eukaryotic genomes**  
**dos Reis M** and Wernisch L. (2009) Molecular Biology and Evolution, 26: 451–461. **[Supporting material](https://href.li/?http://people.cryst.bbk.ac.uk/%7Efdosr01/Nes/index.html)**  
[DOI: 10.1093/molbev/msn272](https://dx.doi.org/10.1093/molbev/msn272)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/molbev/msn272"></div></p>

# 2006

(4) **Archaeology and evolution of transfer RNA genes in the Escherichia coli genome**  
Withers M, Wernisch L, and **dos Reis M**. (2006) RNA, 12: 933–942. **[Supporting material](https://href.li/?http://people.cryst.bbk.ac.uk/%7Efdosr01/trnas/)**  
[DOI: 10.1261/rna.2272306](https://dx.doi.org/10.1261/rna.2272306)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1261/rna.2272306"></div></p>

# 2004

(3) **Solving the riddle of codon usage preferences: a test for translational selection**  
**dos Reis M**, Savva R and Wernisch L. (2004) Nucleic Acids Research, 32: 5036–5044. **[Software](https://href.li/?http://people.cryst.bbk.ac.uk/%7Efdosr01/tAI/)**  
[DOI: 10.1093/nar/gkh834](https://dx.doi.org/10.1093/nar/gkh834)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/nar/gkh834"></div></p>

# 2003

(2) **Churchill, a zinc finger transcriptional activator, regulates the transition between gastrulation and neurulation**  
Sheng G, **dos Reis M**, and Stern CD. (2003) Cell, 115: 603–613. **[Supporting material](http://people.cryst.bbk.ac.uk/~fdosr01/ChCh/ChChSM.pdf) - [F1000 review](https://href.li/?http://f1000.com/prime/1016513)**  
[DOI: 10.1016/S0092-8674(03)00927-9](https://dx.doi.org/10.1016/S0092-8674(03)00927-9)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1016/S0092-8674(03)00927-9"></div></p>

(1) **Unexpected correlations between gene expression and codon usage bias from microarray data for the whole Escherichia coli K-12 genome**  
**dos Reis M**, Wernisch L, and Savva R. (2003) Nucleic Acids Research, 31: 6976–6985.  
[DOI: 10.1093/nar/gkg897](https://dx.doi.org/10.1093/nar/gkg897)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1093/nar/gkg897"></div></p>

# PhD Dissertation

**Understanding the patterns of codon usage bias in prokaryotic and eukaryotic genomes**  
**dos Reis M.** (2010) Birkbeck College, London. 152 pp.  
[Dissertation](/assets/pdfs/dissertation-20100712.pdf)

# Preprints

(5) **A mutation-selection model of protein evolution under persistent positive selection**  
Tamuri AU, and **dos Reis M.** (2021) bioRxiv, 444611.  
[DOI: 10.1101/2021.05.18.444611](https://doi.org/10.1101/2021.05.18.444611)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1101/2021.05.18.444611"></div></p>  

(4) **Pedigree-based measurement of the de novo mutation rate in the gray mouse lemur reveals a high mutation rate, few mutations in CpG sites, and a weak sex bias**  
Campbell CR, Tiley GP, Poelstra JW, Hunnicutt KE, Larsen PA, **dos Reis M**, and Yoder AD. (2019) bioRxiv, 724880.  
[DOI: 10.1101/724880](https://doi.org/10.1101/724880)  
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1101/724880"></div></p>  

(3) **Bayesian estimation of species divergence times using correlated quantitative characters**  
Álvarez-Carretero S, Goswami A, Yang Z, and **dos Reis M.** (2018) bioRxiv, 441105.  
[DOI: 10.1101/441105](https://doi.org/10.1101/441105)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1101/441105"></div></p>

(2)  **Using phylogenomic data to explore the effects of relaxed clocks and calibration strategies on divergence time estimation: Primates as a test case**  
**dos Reis M**, Gunnell GF, Barba-Montoya J, Wilkins A, Yang Z, and Yoder A. (2017) bioRxiv, 201327.  
[DOI: 10.1101/201327](https://doi.org/10.1101/201327)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-doi="10.1101/201327"></div></p>

(1)  **Population genetics and substitution models of adaptive evolution**  
**dos Reis M**. (2013) arXiv, 1311.6682.  
[arXiv: 1311.6682](https://arxiv.org/abs/1311.6682)
<p><div data-badge-popover="right" class='altmetric-embed' data-badge-type='donut' data-arxiv-id="1311.6682"></div></p>

# Peer-review

Associate editor of Genome Biology and Evolution (2020--).

Have reviewed papers for Algorithms for Molecular Biology, Animal Genetics, Bioinformatics, Biology Letters, BioSystems, BMC Evolutionary Biology, Cladistics, Current Biology, eLife, F1000Research, FEBS letters, Gene Expression Patterns, Genome Biology and Evolution, Geobiology, Infection Genetics and Evolution, Interface Focus, Journal of Evolutionary Biology, Journal of Molecular Evolution, Journal of Biogeography, Journal of Theoretical Biology, Mechanisms of Development, Molecular Biology and Evolution, Molecular Ecology Resources, Molecular Genetics and Genomics, Molecular Phylogenetics and Evolution, Nature Communications, Nature Ecology and Evolution, Nature Methods, Nucleic Acids Research, PeerJ, Palaeontology, Plant Cell Reports, Philosophical Transactions of the Royal Society B, PLoS Biology, PLoS Computational Biology, PLoS Genetics, PLoS One, Proceedings of the National Academy of Sciences, Proceedings of the Royal Society B, Science Advances, Scientific Reports, Systematic Biology, Trends in Ecology and Evolution, Virus Evolution and Wellcome Open Research.

Have reviewed for The Austrian Science Fund (FWF), The Biotechnology and Biological Sciences Research Council (BBSRC, UK), The Czech Academy of Sciences, The French National Research Agency (ANR), The Ministry of Health of Singapore, The Science and Technology Facilities Research Council (STFC, UK), The UK Research and Innovation Fellowships, and the Wellcome Trust UK.

<script type='text/javascript' src='https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js'></script>
