<!DOCTYPE html>
<html lang="en">

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Publications</title>
  <meta name="description" content="This is the website for the dos Reis lab research group @ QMUL.">

  <link rel="stylesheet" href="/assets/main.css">
  <link rel="canonical" href="http://localhost:4000/publications/">
  <link rel="alternate" type="application/rss+xml" title="dos Reis Lab" href="/feed.xml">
  
  
</head>


  <body>

    <header class="site-header" role="banner">

  <div class="wrapper">

    <a class="site-title" href="/">dos Reis Lab</a>

    <nav class="site-nav">
      <span class="menu-icon">
        <svg viewBox="0 0 18 15" width="18px" height="15px">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </span>

      <div class="trigger">
        
          
          <a class="page-link" href="/about/">About</a>
          
        
          
        
          
          <a class="page-link" href="/publications/">Publications</a>
          
        
          
        
          
        
          
        
      </div>
    </nav>

  </div>

</header>


    <main class="page-content" aria-label="Content">
      <div class="wrapper">
        <article class="post">

  <header class="post-header">
    <h1 class="post-title">Publications</h1>
  </header>

  <div class="post-content">
    <p>Publications
Google Scholar profile: <a href="http://scholar.google.co.uk/citations?user=zy5V3qMAAAAJ&amp;hl">scholar.google.co.uk/citations?user=zy5V3qMAAAAJ&amp;hl</a></p>

<h1 id="2017">2017</h1>

<ol>
  <li>
    <p><strong>A biologist’s guide to Bayesian phylogenetic analysis</strong><br />
Nascimento FF, <strong>dos Reis M</strong>, and Yang Z. (2017) Nature Ecology and Evolution, 1: 1446–1454.<br />
<a href="https://dx.doi.org/10.1038/s41559-017-0280-x">DOI: 10.1038/s41559-017-0280-x</a></p>
  </li>
  <li>
    <p><strong>Comparison of different strategies for using fossil calibrations to generate the time prior in Bayesian molecular clock dating.</strong><br />
Barba-Montaya J, <strong>dos Reis M</strong>, and Yang Z. (2017) Molecular Phylogenetics and Evolution, 114: 386–400.<br />
<a href="https://dx.doi.org/10.1016/j.ympev.2017.07.005">DOI: 10.1016/j.ympev.2017.07.005</a></p>
  </li>
  <li>
    <p><strong>An evaluation of different partitioning strategies for Bayesian estimation of species divergence times.</strong><br />
Konstantinos A, Álvarez-Carretero A, <strong>dos Reis M</strong> and Yang Z. (2017) Systematic Biology (in the press).<br />
<a href="https://dx.doi.org/10.1093/sysbio/syx061">DOI: 10.1093/sysbio/syx061</a></p>
  </li>
  <li>
    <p><strong>RelTime rates collapse to a strict clock when estimating the timeline of animal diversification.</strong><br />
Lozano Fernandez J, <strong>dos Reis M</strong>, Donoghue PCJ and Pisani D. (2017) Genome Biology and Evolution, 9: 1320–1328.<br />
<a href="https://dx.doi.org/10.1093/gbe/evx079">DOI: 10.1093/gbe/evx079</a></p>
  </li>
  <li>
    <p><strong>Finding direction in the search for selection.</strong><br />
Thiltgen G, <strong>dos Reis M</strong> and Goldstein RA. (2017) Journal of Molecular Evolution, 84: 39–50.<br />
<a href="https://dx.doi.org/10.1007/s00239-016-9765-5">DOI: 10.1007/s00239-016-9765-5</a> Data: <a href="https://dx.doi.org/10.6084/m9.figshare.4284380.v1">DOI: 10.6084/m9.figshare.4284380.v1</a></p>
  </li>
</ol>

<h1 id="2016">2016</h1>

<ol>
  <li>
    <p><strong>Geogenetic patterns in mouse lemurs (genus Microcebus) reveal the ghosts of Madagascar’s forests past.</strong><br />
Yoder AD, Campbell CR, Blanco MB, <strong>dos Reis M</strong>, Ganzhorn JU, Goodman SM, Hunnicutt KE, Larsen PA, Kappeler PM, Rasoloarison RM, Ralison JM, Swofford DL and Weisrock DW. (2016) Proceedings of the National Academy of Sciences, 113: 8049–8056. <strong>[comments in The Washington Post and National Geographic]</strong><br />
<a href="https://dx.doi.org/10.1073/pnas.1601081113">DOI: 10.1073/pnas.1601081113</a></p>
  </li>
  <li>
    <p><strong>Notes on the birth-death prior with fossil calibrations for Bayesian estimation of species divergence times.</strong><br />
<strong>dos Reis M</strong>. (2016) Philosophical Transactions of the Royal Society B, 371: 20150128.<br />
<a href="https://dx.doi.org/10.1098/rstb.2015.0128">DOI: 10.1098/rstb.2015.0128</a></p>
  </li>
  <li>
    <p><strong>The Interrelationships of placental mammals and the limits of phylogenetic inference.</strong><br />
Tarver JE, <strong>dos Reis M</strong>,  Mirarab S, Moran RJ, Parker S, O’Reilly JE, King BL, O’Connell MJ, Asher RJ, Warnow T, Peterson KJ, Donoghue PCJ and Pisani D. (2016) Genome Biology and Evolution, 8: 330–344.<br />
<a href="https://dx.doi.org/10.1093/gbe/evv261">DOI: 10.1093/gbe/evv261</a></p>
  </li>
  <li>
    <p><strong>Bayesian molecular clock dating of species divergences in the genomics era.</strong><br />
<strong>dos Reis M</strong>, Donoghue PCJ and Yang Z. (2016) Nature Reviews Genetics, 17: 71–80.<br />
<a href="https://dx.doi.org/10.1038/nrg.2015.8">DOI: 10.1038/nrg.2015.8</a></p>
  </li>
</ol>

<h1 id="2015">2015</h1>

<ol>
  <li>
    <p><strong>Uncertainty in the timing of origin of animals and the limits of precision in molecular timescales.</strong><br />
<strong>dos Reis M</strong>, Thawornwattana Y, Angelis K, Telford M, Donoghue PCJ and Yang Z. (2015) Current Biology, 25: 2939–2950. <strong>[featured article]</strong><br />
<a href="https://dx.doi.org/10.1016/j.cub.2015.09.066">DOI: 10.1016/j.cub.2015.09.066</a> Data: <a href="https://dx.doi.org/10.6084/m9.figshare.1525089">DOI: 10.6084/m9.figshare.1525089</a></p>
  </li>
  <li>
    <p><strong>Dating tips for divergence time estimation.</strong><br />
O’Reilly J, <strong>dos Reis M</strong> and Donoghue PCJ. (2015) Trends in Genetics, 31: 637–650.<br />
<a href="https://dx.doi.org/10.1016/j.tig.2015.08.001">DOI: 10.1016/j.tig.2015.08.001</a></p>
  </li>
  <li>
    <p><strong>The impact of ancestral population size and incomplete lineage sorting on Bayesian estimation of species divergence times.</strong><br />
Angelis K and <strong>dos Reis M</strong>. (2015) Current Zoology, 61: 874–885.<br />
<a href="https://dx.doi.org/10.1093/czoolo/61.5.874">DOI: 10.1093/czoolo/61.5.874</a></p>
  </li>
  <li>
    <p><strong>How to calculate the non-synonymous to synonymous rate ratio of protein-coding coding genes under the Fisher-Wright mutation-selection framework.</strong><br />
<strong>dos Reis M</strong>. (2015) Biology Letters, 11: 20141031.<br />
<a href="https://dx.doi.org/10.1098/rsbl.2014.1031">DOI: 10.1098/rsbl.2014.1031</a> 
Data: <a href="https://dx.doi.org/10.5061/dryad.3r3q4">DOI: 10.5061/dryad.3r3q4</a> 
R code and examples: <a href="https://dx.doi.org/10.6084/m9.figshare.4282835.v1">DOI: 10.6084/m9.figshare.4282835.v1</a></p>
  </li>
  <li>
    <p><strong>Characterization of the uncertainty of divergence time estimation under relaxed molecular clock models using multiple loci.</strong><br />
Zhu T, <strong>dos Reis M</strong> and Yang Z. (2015) Systematic Biology, 64: 267–280.<br />
<a href="https://dx.doi.org/10.1093/sysbio/syu109">DOI: 10.1093/sysbio/syu109</a></p>
  </li>
  <li>
    <p><strong>A beginner’s guide to estimating the non-synonymous to synonymous rate ratio of all protein-coding genes in a genome.</strong><br />
Jeffares DC, Tomiczek B, Sojo V and <strong>dos Reis M</strong>. (2015) Parasite Genomics Protocols, Methods in Molecular Biology, 1201: 65–90.<br />
<a href="https://dx.doi.org/10.1007/978-1-4939-1438-8_4">DOI: 10.1007/978-1-4939-1438-8_4</a></p>
  </li>
</ol>

<h1 id="2014">2014</h1>

<ol>
  <li>
    <p><strong>Bayesian estimation of nonsynonymous/synonymous rate ratios for pairwise sequence comparisons.</strong><br />
Angelis, K. <strong>dos Reis M</strong> and Yang Z. (2014) Molecular Biology and Evolution, 31: 1902–1913.<br />
<a href="https://dx.doi.org/10.1093/molbev/msu142">DOI: 10.1093/molbev/msu142</a></p>
  </li>
  <li>
    <p><strong>The impact of the rate prior on Bayesian estimation of divergence times with multiple loci.</strong><br />
<strong>dos Reis M</strong>, Zhu T and Yang Z. (2014) Systematic Biology, 63: 555–565.<br />
<a href="https://dx.doi.org/10.1093/sysbio/syu020">DOI: 10.1093/sysbio/syu020</a></p>
  </li>
  <li>
    <p><strong>A penalized-likelihood method to estimate the distribution of selection coefficients from phylogenetic data.</strong><br />
Tamuri AU, Goldman N and <strong>dos Reis M</strong>. (2014) Genetics, 197: 257–271. <strong>[highlighted article]</strong><br />
<a href="https://dx.doi.org/10.1534/genetics.114.162263">DOI: 10.1534/genetics.114.162263</a></p>
  </li>
  <li>
    <p><strong>Molecular evolutionary characterization of a V1R subfamily unique to strepsirrhine Primates.</strong><br />
Yoder AD, Chan LM, <strong>dos Reis M</strong>, Larsen PA, Campbell CR, Rasolarison R, Barrett M, Roos C, Kappeler P, Bielawski JP and Yang Z. (2014) Genome Biology and Evolution, 6: 213–227.<br />
<a href="https://dx.doi.org/10.1093/gbe/evu006">DOI: 10.1093/gbe/evu006</a></p>
  </li>
  <li>
    <p><strong>Neither phylogenomic nor palaeontological data support a Paleogene origin of placental mammals.</strong><br />
<strong>dos Reis M</strong>, Donoghue PCJ and Yang Z. (2014) Biology Letters, 10: 20131003. <strong>[comments in Nature, The Scientist, Phys.org and Discovery News] [F1000 review]</strong><br />
<a href="https://dx.doi.org/10.1098/rsbl.2013.1003">DOI: 10.1098/rsbl.2013.1003</a></p>
  </li>
</ol>

<h1 id="2013">2013</h1>

<ol>
  <li>
    <p><strong>Why do more divergent sequences produce smaller nonsynonymous/synonymous rate ratios in pairwise sequence comparisons?</strong><br />
<strong>dos Reis M</strong> and Yang Z. (2013) Genetics, 195: 195–204.<br />
<a href="https://dx.doi.org/10.1534/genetics.113.152025">DOI: 10.1534/genetics.113.152025</a></p>
  </li>
  <li>
    <p><strong>The unbearable uncertainty of Bayesian divergence time estimation.</strong><br />
<strong>dos Reis M</strong> and Yang Z. (2013) Journal of Systematics and Evolution, 51: 30–43.<br />
<a href="https://dx.doi.org/10.1111/j.1759-6831.2012.00236.x">DOI: 10.1111/j.1759-6831.2012.00236.x</a></p>
  </li>
</ol>

<h1 id="2012">2012</h1>

<ol>
  <li>
    <p><strong>Phylogenomic datasets provide both precision and accuracy in estimating the timescale of placental mammal phylogeny.</strong><br />
<strong>dos Reis M</strong>, Inoue J, Hasegawa M, Asher RJ, Donoghue PCJ and Yang Z. (2012) Proceedings of the Royal Society B., 279: 3491–3500. <strong>[timetrees (nexus)] [274 species timetree (pdf)] [data] [F1000 review]</strong><br />
<a href="https://dx.doi.org/10.1098/rspb.2012.0683">DOI: 10.1098/rspb.2012.0683</a></p>
  </li>
  <li>
    <p><strong>Estimating the distribution of selection coefficients from phylogenetic data using sitewise mutation-selection models.</strong><br />
Tamuri AU, <strong>dos Reis M</strong> and Goldstein RA. (2012) Genetics, 190: 1101–1115.<br />
<a href="https://dx.doi.org/10.1534/genetics.111.136432">DOI: 10.1534/genetics.111.136432</a></p>
  </li>
</ol>

<h1 id="2011">2011</h1>

<ol>
  <li>
    <p><strong>An exceptionally high nucleotide and haplotype diversity and a signature of positive selection for the eIF4E resistance gene in barley are revealed by allele mining and phylogenetic analyses of natural populations.</strong><br />
Hofinger BJ, Russell JR, Bass CG, Baldwin T, <strong>dos Reis M</strong>, Hedley PE, Li Y, Macaulay M, Waugh R, Hammond-Kosack KE and Kanyuka K. (2011) Molecular Ecology, 20: 3653–3668.<br />
<a href="https://dx.doi.org/10.1111/j.1365-294X.2011.05201.x">DOI: 10.1111/j.1365-294X.2011.05201.x</a></p>
  </li>
  <li>
    <p><strong>Approximate likelihood calculation on a phylogeny for Bayesian estimation of divergence times.</strong><br />
<strong>dos Reis M</strong> and Yang Z. (2011) Molecular Biology and Evolution, 28: 2161–2172. <strong>[R code]</strong><br />
<a href="https://dx.doi.org/10.1093/molbev/msr045">DOI: 10.1093/molbev/msr045</a></p>
  </li>
  <li>
    <p><strong>Charting the host adaptation of influenza viruses.</strong><br />
<strong>dos Reis M</strong>, Tamuri AU, Hay AJ and Goldstein RA. (2011) Molecular Biology and Evolution, 28: 1755–1767. <strong>[featured in Evolutionary Analysis]</strong><br />
<a href="https://dx.doi.org/10.1093/molbev/msq317">DOI: 10.1093/molbev/msq317</a></p>
  </li>
  <li>
    <p><strong>Statistical properties of the branch-site test of positive selection.</strong><br />
Yang Z and <strong>dos Reis M</strong>. (2011) Molecular Biology and Evolution, 28: 1217–1228.<br />
<a href="https://dx.doi.org/10.1093/molbev/msq303">DOI: 10.1093/molbev/msq303</a></p>
  </li>
</ol>

<h1 id="2010">2010</h1>

<ol>
  <li><strong>Analysis of the Albumin/alpha-Fetoprotein/Afamin/Group specific component gene family in the context of zebrafish liver differentation.</strong><br />
Noël ES, <strong>dos Reis M</strong>, Arain Z and Ober EA. (2010) Gene Expression Patterns, 10: 237–243.<br />
<a href="https://dx.doi.org/10.1016/j.gep.2010.05.002">DOI: 10.1016/j.gep.2010.05.002</a></li>
</ol>

<h1 id="2009">2009</h1>

<ol>
  <li>
    <p><strong>Identifying changes in selective constraints: host shifts in influenza.</strong><br />
Tamuri AU, <strong>dos Reis M</strong>, Hay AJ and Goldstein RA. (2009) PLoS Computational Biology, 5: e1000564.<br />
<a href="https://dx.doi.org/10.1371/journal.pcbi.1000564">DOI: 10.1371/journal.pcbi.1000564</a></p>
  </li>
  <li>
    <p><strong>Using non-homogeneous models to identify host shift events: application to the origin of the 1918 ‘Spanish’ influenza virus.</strong><br />
<strong>dos Reis M</strong>, Hay AJ and Goldstein RA. (2009) Journal of Molecular Evolution, 69: 333–345. <strong>[Additional data] [featured in Evolutionary Analysis]</strong><br />
<a href="https://dx.doi.org/10.1007/s00239-009-9282-x">DOI: 10.1007/s00239-009-9282-x</a></p>
  </li>
  <li>
    <p><strong>Estimating translational selection in eukaryotic genomes.</strong><br />
<strong>dos Reis M</strong> and Wernisch L. (2009) Molecular Biology and Evolution, 26: 451–461. <strong>[Supporting material]</strong><br />
<a href="https://dx.doi.org/10.1093/molbev/msn272">DOI: 10.1093/molbev/msn272</a></p>
  </li>
</ol>

<h1 id="2006">2006</h1>

<ol>
  <li><strong>Archaeology and evolution of transfer RNA genes in the Escherichia coli genome.</strong><br />
Withers M, Wernisch L, and <strong>dos Reis M</strong>. (2006) RNA, 12: 933–942. <strong>[Supporting material]</strong><br />
<a href="https://dx.doi.org/10.1261/rna.2272306">DOI: 10.1261/rna.2272306</a></li>
</ol>

<h1 id="2004">2004</h1>

<ol>
  <li><strong>Solving the riddle of codon usage preferences: a test for translational selection.</strong><br />
<strong>dos Reis M</strong>, Savva R and Wernisch L. (2004) Nucleic Acids Research, 32: 5036–5044. <strong>[Software]</strong><br />
<a href="https://dx.doi.org/10.1093/nar/gkh834">DOI: 10.1093/nar/gkh834</a></li>
</ol>

<h1 id="2003">2003</h1>

<ol>
  <li>
    <p><strong>Churchill, a zinc finger transcriptional activator, regulates the transition between gastrulation and neurulation.</strong><br />
Sheng G, <strong>dos Reis M</strong>, and Stern CD. (2003) Cell, 115: 603–613. <strong>[Supporting material] [F1000 review]</strong><br />
<a href="https://dx.doi.org/10.1016/S0092-8674(03)00927-9">DOI: 10.1016/S0092-8674(03)00927-9</a></p>
  </li>
  <li>
    <p><strong>Unexpected correlations between gene expression and codon usage bias from microarray data for the whole Escherichia coli K-12 genome.</strong><br />
<strong>dos Reis M</strong>, Wernisch L, and Savva R. (2003) Nucleic Acids Research, 31: 6976–6985.<br />
<a href="https://dx.doi.org/10.1093/nar/gkg897">DOI: 10.1093/nar/gkg897</a></p>
  </li>
</ol>

<h1 id="other">Other</h1>

<ol>
  <li><strong>Population genetics and substitution models of adaptive evolution.</strong><br />
<strong>dos Reis M</strong>. (2013) arXiv.<br />
<a href="https://arxiv.org/abs/1311.6682">arXiv: 1311.6682</a></li>
</ol>

<p>I have acted as a reviewer for Algorithms for Molecular Biology, Animal Genetics, Bioinformatics, Biology Letters, BioSystems, BMC Evolutionary Biology, Cladistics, Current Biology, FEBS letters, Gene Expression Patterns, Geobiology, Infection Genetics and Evolution, Journal of Theoretical Biology, Mechanisms of Development, Molecular Biology and Evolution, Molecular Ecology Resources, Molecular Genetics and Genomics, Molecular Phylogenetics and Evolution, Nature Ecology and Evolution, Nucleic Acids Research, PeerJ, Plant Cell Reports, Philosophical Transactions of the Royal Society B, PLoS Computational Biology, PLoS Genetics, PLoS One, Proceedings of the National Academy of Sciences, Proceedings of the Royal Society B, Science Advances, Systematic Biology, and Trends in Ecology and Evolution. Have reviewed grants for The Austrian Science Fund (FWF), The French National Research Agency (ANR), The Ministry of Health of Singapore, and the Wellcome Trust UK.</p>

  </div>

</article>

      </div>
    </main>

    <footer class="site-footer">

  <div class="wrapper">

    <h2 class="footer-heading">dos Reis Lab</h2>

    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li>
            
              dos Reis Lab
            
            </li>
            
            <li><a href="mailto:m.dosreisbarros@qmul.ac.uk">m.dosreisbarros@qmul.ac.uk</a></li>
            
        </ul>
      </div>

      <div class="footer-col footer-col-2">
        <ul class="social-media-list">
          
          <li>
            <a href="https://github.com/mariodosreis"><span class="icon icon--github"><svg viewBox="0 0 16 16" width="16px" height="16px"><path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/></svg>
</span><span class="username">mariodosreis</span></a>

          </li>
          

          
          <li>
            <a href="https://twitter.com/mariodosreis"><span class="icon icon--twitter"><svg viewBox="0 0 16 16" width="16px" height="16px"><path fill="#828282" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27 c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767 c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206 C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271 c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469 c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/></svg>
</span><span class="username">mariodosreis</span></a>

          </li>
          
        </ul>
      </div>

      <div class="footer-col footer-col-3">
        <p>This is the website for the dos Reis lab research group @ QMUL.
</p>
      </div>
    </div>

  </div>

</footer>


  </body>

</html>
