---
layout: post
title:  "Workshop: Bayesian methods to estimate species divergence times"
author: "Mario dos Reis"
---

Together with Ziheng Yang and Philip Donoghue I am organising a workshop on Bayesian phylogenetic methods. The workshop will take place between the 30th and 31st of July of 2015 at the University of Bristol. To register please visit [bit.ly/1INFQxZ](http://bit.ly/1INFQxZ). This page will be updated regularly so stay tuned!

# Summary

In this two-day workshop we will introduce the theory and practice of Bayesian statistical methods to analyze molecular, morphological and fossil data to estimate the times of divergence of species on a phylogeny. The workshop is aimed at scientists (graduate students, postdocs and established academics) interested in using the methods in their own research. Attendees will learn how to use R to write a basic Baysian phylogenetics program, and will learn how to use specialized packages such as Beast, MCMCTree and MrBayes for divergence time estimation. The workshop will introduce attendees to the newer methods for combined analysis of morphological and molecular data (so-called total evidence dating). In particular, there may be an opportunity for attendees to learn how to analyse continuous morphological data (such as landmarks) with the program MCMCTree. The workshop is made possible thanks to funding by the BBSRC, UK.

# Prerequisites

Attendees are required to have basic knowledge of statistics and basic knowledge of the R computer package for statistics (for example, you know how to perform basic calculations with R, know how to define functions and how to write ‘for loops’). Attendees should ideally have some basic experience in phylogenetics (for example, you may have used a phylogenetics program such as RAxML to build a phylogenetic tree). Basic knowledge of the command line or terminal (in Windows, Mac or Linux) is also necessary. Attendees must bring their own laptops to participate in the workshop. Detailed instructions about the software the attendees need to install in their laptops will be given close to the workshop date.

# Programme

Thursday 30th
- 09:00-10:00 Registration. Tea and coffee.
- 10:00-11:00 Lecture: Phil Donoghue on the fossil record.
- 11:00-12:00 Lecture: Ziheng Yang on Bayesian phylogenetics theory.
- 12:00-13:30 Lunch.
- 13:30-15:30 R workshop: How to write your own MCMC phylogenetics program.
- 15:30-16:00 Tea and coffee break.
- 16:00-18:00 R workshop: MCMC diagnostics.
- 18:00-19:00 Wine reception.

Friday 31st
- 09:00-10:00 Questions and Answers session. Tea and Coffee.
- 10:00-12:00 Beast workshop: Dating a virus phylogeny.
- 12:00-13:30 lunch.
- 13:30-15:30 MCMCTree workshop: Analysis of genome-scale data.
- 15:30-16:00 Tea and coffee break.
- 16:00-18:00 MrBayes workshop: Combined analysis of morphological and molecular data.

# Further Notes

- You will need to bring your own laptop, and install several computer packages on it, in order to participate in the workshop. You are expected to be familiar using the command line (a.k.a. terminal, shell, etc.) of the operating system in your laptop (whether it is Windows or a Unix-like system such as Mac OS X or Linux). If you need to polish your command line skills, you can find various tutorials [here](http://abacus.gene.ucl.ac.uk/software.html).

- On the first day there will be a session where you will write your own R program to perform Bayesian MCMC phylogenetics. Thus you are expected to be familiar with the R environment for statistical computing. An extensive introduction to R can be found [here](http://cran.r-project.org/manuals.html). You can also check my own short R tutorial [here](/2015/02/19/crash-r-intro.html). Make sure you have installed the latest R version.

- You need to install the PAML v4.8a package, which contains the MCMCTree program to be used in the workshop. Ideally you would set the Path variable so the the operating system can find the PAML executables. Instructions for Windows vista are in the PAML’s website. Linux and Mac OS X users should search online for setting $PATH.

- You need to install MrBayes v3.2.5. You may want to add the mb executable to your Path variable as well.

- You need to install Beast v1.8.2, Tracer v1.6, and FigTree v1.4.2. These programs need Java installed in your system. My understanding is that there are two versions of Beast being currently developed in parallel: 1.8 and 2. We will work with 1.8 only, so make sure you have installed the right version, as there are substantial differences between 1.8 and 2.

- Please install the software needed well in advance, and do some basic tests to make sure that the programs work correctly in your laptop. Our ability to help people out with software installation issues during the workship will be very limited.

- BOOK: An easy to read, yet extensive, introduction to Bayesian statistics and Bayesian phylogenetics can be found in chapters six to eight of “Molecular evolution: A statistical approach“.

- FOOD: We will provide tea and coffee and lunch on both days at no extra cost. If you have any special dietary requirements please contact me as soon as possible.

- CANCELLATIONS: If due to unforeseen circumstances you cannot make the workshop, please let us know a.s.a.p so that we can offer your place to somebody else.

For enquires contact: Mario dos Reis, mario.barros@ucl.ac.uk
