---
layout: post
title:   "Simple analysis"
author: "Mario dos Reis"
---

A nucleotide alignment of 16 species was simulated on the phylogeny of Figure
1A. The model is GTR+G with nucleotide frequencies T = 0.25318,  C = 0.32894,
A = 0.31196,  G = 0.10592; alpha = 0.25; and substitution matrix parameters
equal to 0.88892,  0.03190,  0.00001,  0.07102,  0.02418. Simulation software is
Evolver v4.9e. Mean substitution rate is 1.

The data are then analysed with MCMCtree v4.9f to recover the divergence times under
the JC69 and GTR+G substitution models using the approximate likelihood
method.

In Figure 1B, posterior mean times estimated under JC69 are shown in the y axis,
while corresponding estimates for GTR+G are shown in the x axis. Times are
estimated using a precise calibration on the
youngest node in the phylogeny (red dot) and a very diffuse calibration on the root. The
root calibration is G(2, 0.66) which has mean 3.03 (close to the true root age)
and 95% prior CI (0.367, 8.44). In this case, **there is catastrophic underestimation
error when using the JC69 model** (which is misspecified) in comparison to the
GTR+G model (which is the correct model).

In Figure 1C, the precise calibration is moved to the sixth youngest node in the
phylogeny (red dot). The effect of sliding the precise calibration to an older node is to
"pull up" the estimated times under JC69 so that they are in better agreement
with the estimates under GTR+G. However, there is still unacceptably large underestimation
error for the ages of the oldest nodes.

In Figure 1D, the precise calibration has been moved to the root (red dot). Here both JC69
and GTR+G show broadly similar time estimates. In other words, a precise calibration
on the root pins the inferred timespan of the phylogeny making the analysis robust
to substitution model misspecification.

Figure 1:
![](/assets/figs/timesJC69vGTRG.png)
