---
layout: post
title:   "Simple analysis"
author: "Mario dos Reis"
---

A nucleotide alignment of 16 species was simulated on the phylogeny of Figure
1A. The model is GTR+G with nucleotide frequencies T = 0.25318,  C = 0.32894,
A = 0.31196,  G = 0.10592; alpha = 0.25; and substitution matrix parameters
equal to 0.88892,  0.03190,  0.00001,  0.07102,  0.02418. Simulation software is
Evolver v4.9e. Mean substitution rate is 1.

The data are then analysed with MCMCtree v4.9f to recover the divergence times under
the JC69 and GTR+G substitution models using the approximate likelihood
method and the strict clock.

In Figure 1B, posterior mean times estimated under JC69 are shown in the y axis,
while corresponding estimates for GTR+G are shown in the x axis. Times are
estimated using a precise calibration on the
youngest node in the phylogeny (red dot) and a very diffuse calibration on the root. The
root calibration is G(2, 0.66) which has mean 3.03 (close to the true root age)
and 95% prior CI (0.367, 8.44). In this case, **there is catastrophic underestimation
error when using the JC69 model** (which is misspecified) in comparison to the
GTR+G model (which is the correct model).

In Figure 1C, the precise calibration is moved to the sixth youngest node in the
phylogeny (red dot). The effect of sliding the precise calibration to an older node is to
"pull up" the estimated times under JC69 so that they are in better agreement
with the estimates under GTR+G. However, there is still unacceptably large underestimation
error for the ages of the oldest nodes.

In Figure 1D, the precise calibration has been moved to the root (red dot). Here both JC69
and GTR+G show broadly similar time estimates. In other words, a precise calibration
on the root pins the inferred timespan of the phylogeny making the analysis robust
to substitution model misspecification.

**Figure 1:**

![](/assets/figs/timesJC69vGTRG.png)

**Appendix:**

Phylogeny used in the simulation:

```
(a:3.0,(b:2.8,(c:2.6,(d:2.4,(e:2.2,(f:2.0,(g:1.8,(h:1.6,(i:1.4,(j:1.2,(k:1.0,
  (l:0.8,(m:0.6,(n:0.4,(o:0.2,p:0.2):0.2):0.2):0.2):0.2):0.2):0.2):0.2):0.2):0.2
  ):0.2):0.2):0.2):0.2):0.2);
```

Tree with fossil calibrations used:
```
# Informative calibration on youngest node:
16 1

(a,(b,(c,(d,(e,(f,(g,(h,(i,(j,(k,(l,(m,(n,(o,p)'B(.18,.22)'))))))))))))))'G(2,.66)';

# Informative calibration on mid node:
16 1

(a,(b,(c,(d,(e,(f,(g,(h,(i,(j,(k,(l,(m,(n,(o,p))))))'B(1.08,1.32)')))))))))'G(2,.66)';

# Informative calibration on root node:
16 1

(a,(b,(c,(d,(e,(f,(g,(h,(i,(j,(k,(l,(m,(n,(o,p)))))))))))))))'B(2.7,3.3)';
```

Time estimates (the true ages for the 15 nodes are 0.2, 0.4, 0.6, ..., 2.6, 2.8, 3.0):

```
# Informative calibration on youngest node - JC69:
node t t025 t975 t025.hpd t975.hpd w.hpd
t_n17 0.9991 0.8160 1.2175 0.8068 1.2029 0.3960
t_n18 0.9846 0.8059 1.1990 0.7941 1.1811 0.3870
t_n19 0.9554 0.7806 1.1624 0.7723 1.1498 0.3774
t_n20 0.8718 0.7136 1.0582 0.7001 1.0399 0.3398
t_n21 0.8230 0.6746 0.9973 0.6639 0.9840 0.3201
t_n22 0.7496 0.6145 0.9086 0.6097 0.9001 0.2904
t_n23 0.7059 0.5772 0.8552 0.5727 0.8486 0.2760
t_n24 0.6410 0.5247 0.7772 0.5187 0.7697 0.2510
t_n25 0.5806 0.4756 0.7021 0.4674 0.6920 0.2246
t_n26 0.5339 0.4378 0.6452 0.4329 0.6372 0.2044
t_n27 0.4885 0.4010 0.5881 0.3984 0.5848 0.1865
t_n28 0.4252 0.3501 0.5101 0.3459 0.5046 0.1588
t_n29 0.3723 0.3080 0.4444 0.3058 0.4412 0.1354
t_n30 0.3002 0.2521 0.3531 0.2494 0.3493 0.0999
t_n31 0.2057 0.1816 0.2208 0.1829 0.2214 0.0385

# Informative calibration on youngest node - GTR+G:
node t t025 t975 t025.hpd t975.hpd w.hpd
t_n17 3.5934 2.7091 4.7786 2.6343 4.6371 2.0029
t_n18 3.3596 2.5708 4.3645 2.4917 4.2579 1.7662
t_n19 3.1178 2.3844 4.0521 2.3242 3.9596 1.6354
t_n20 2.9652 2.2758 3.8480 2.2114 3.7455 1.5341
t_n21 2.8242 2.1690 3.6543 2.1355 3.5898 1.4543
t_n22 2.4809 1.9058 3.2078 1.8807 3.1547 1.2741
t_n23 2.3524 1.8073 3.0430 1.7701 2.9820 1.2119
t_n24 1.9971 1.5198 2.6110 1.4753 2.5470 1.0717
t_n25 1.7999 1.3730 2.3360 1.3421 2.2895 0.9474
t_n26 1.4235 1.0848 1.8502 1.0614 1.8171 0.7557
t_n27 1.2795 0.9724 1.6617 0.9339 1.6166 0.6827
t_n28 0.9322 0.7040 1.2203 0.6879 1.1984 0.5106
t_n29 0.7710 0.5842 1.0131 0.5700 0.9884 0.4184
t_n30 0.4750 0.3608 0.6225 0.3518 0.6111 0.2593
t_n31 0.2088 0.1846 0.2212 0.1872 0.2222 0.0350

# Informative calibration on mid node - JC69:
node t t025 t975 t025.hpd t975.hpd w.hpd
t_n17 2.2581 1.9377 2.5930 1.9339 2.5854 0.6516
t_n18 2.2257 1.9127 2.5521 1.9025 2.5403 0.6378
t_n19 2.1597 1.8521 2.4801 1.8478 2.4736 0.6258
t_n20 1.9728 1.7014 2.2558 1.6949 2.2476 0.5527
t_n21 1.8635 1.6140 2.1244 1.6112 2.1213 0.5101
t_n22 1.6975 1.4730 1.9288 1.4721 1.9270 0.4549
t_n23 1.5981 1.3912 1.8098 1.3887 1.8064 0.4177
t_n24 1.4514 1.2696 1.6332 1.2654 1.6271 0.3617
t_n25 1.3160 1.1609 1.4649 1.1603 1.4636 0.3032
t_n26 1.2115 1.0821 1.3220 1.0840 1.3238 0.2397
t_n27 1.1093 0.9737 1.2351 0.9772 1.2378 0.2606
t_n28 0.9675 0.8378 1.0925 0.8391 1.0937 0.2546
t_n29 0.8492 0.7293 0.9720 0.7252 0.9675 0.2423
t_n30 0.6901 0.5801 0.8042 0.5796 0.8035 0.2239
t_n31 0.4825 0.3935 0.5797 0.3885 0.5741 0.1856

# Informative calibration on mid node - GTR+G:
node t t025 t975 t025.hpd t975.hpd w.hpd
t_n17 3.3606 2.7580 4.1452 2.7173 4.0630 1.3457
t_n18 3.1511 2.6142 3.7574 2.5983 3.7391 1.1408
t_n19 2.9298 2.4335 3.4957 2.4152 3.4757 1.0605
t_n20 2.7855 2.3184 3.3119 2.3184 3.3111 0.9927
t_n21 2.6532 2.2065 3.1520 2.1884 3.1276 0.9392
t_n22 2.3336 1.9461 2.7630 1.9329 2.7439 0.8110
t_n23 2.2038 1.8437 2.6039 1.8242 2.5813 0.7572
t_n24 1.8590 1.5435 2.2238 1.5370 2.2129 0.6759
t_n25 1.6679 1.3916 1.9913 1.3793 1.9724 0.5930
t_n26 1.2592 1.1174 1.3278 1.1375 1.3368 0.1993
t_n27 1.1633 0.9892 1.2929 1.0084 1.3045 0.2961
t_n28 0.8623 0.7148 1.0111 0.7147 1.0109 0.2962
t_n29 0.7301 0.5915 0.8752 0.5868 0.8688 0.2820
t_n30 0.4634 0.3679 0.5713 0.3625 0.5654 0.2029
t_n31 0.2442 0.1834 0.3130 0.1811 0.3096 0.1285

# Informative calibration on root node - JC69:
node t t025 t975 t025.hpd t975.hpd w.hpd
t_n17 2.9813 2.6993 3.2943 2.6998 3.2945 0.5947
t_n18 2.9367 2.6444 3.2627 2.6545 3.2708 0.6163
t_n19 2.8487 2.5466 3.1816 2.5493 3.1833 0.6339
t_n20 2.5993 2.3034 2.9224 2.3021 2.9209 0.6189
t_n21 2.4535 2.1661 2.7722 2.1585 2.7608 0.6023
t_n22 2.2328 1.9662 2.5335 1.9570 2.5227 0.5657
t_n23 2.1027 1.8445 2.3940 1.8448 2.3940 0.5492
t_n24 1.9095 1.6729 2.1825 1.6616 2.1680 0.5065
t_n25 1.7305 1.5108 1.9822 1.4972 1.9669 0.4697
t_n26 1.5924 1.3850 1.8298 1.3792 1.8199 0.4408
t_n27 1.4579 1.2614 1.6823 1.2590 1.6791 0.4200
t_n28 1.2715 1.0925 1.4769 1.0843 1.4658 0.3815
t_n29 1.1167 0.9546 1.3015 0.9498 1.2949 0.3451
t_n30 0.9068 0.7631 1.0717 0.7535 1.0598 0.3063
t_n31 0.6343 0.5183 0.7672 0.5135 0.7603 0.2468

# Informative calibration on root node - GTR+G:
node t t025 t975 t025.hpd t975.hpd w.hpd
t_n17 3.1422 2.7822 3.3201 2.8271 3.3436 0.5165
t_n18 3.0115 2.6293 3.2711 2.6725 3.2919 0.6194
t_n19 2.8250 2.4328 3.1493 2.4523 3.1620 0.7096
t_n20 2.6991 2.3154 3.0326 2.3488 3.0643 0.7155
t_n21 2.5795 2.2041 2.9185 2.2097 2.9226 0.7129
t_n22 2.2769 1.9392 2.5922 1.9389 2.5915 0.6526
t_n23 2.1582 1.8365 2.4693 1.8412 2.4733 0.6321
t_n24 1.8337 1.5373 2.1385 1.5341 2.1348 0.6007
t_n25 1.6664 1.3800 1.9604 1.3685 1.9485 0.5800
t_n26 1.3248 1.0894 1.5760 1.0787 1.5637 0.4850
t_n27 1.1889 0.9751 1.4170 0.9722 1.4132 0.4410
t_n28 0.8706 0.7070 1.0516 0.7012 1.0426 0.3414
t_n29 0.7257 0.5865 0.8734 0.5851 0.8705 0.2853
t_n30 0.4599 0.3634 0.5661 0.3610 0.5631 0.2021
t_n31 0.2431 0.1819 0.3125 0.1806 0.3108 0.1302
```
